supplementary/alica_ros1/alica_tests/src/test/test_alica_simple_plan.cpp

```diff
@@ -21,60 +21,106 @@
 
 #include <gtest/gtest.h>
 
-namespace alica
-{
-namespace
+#include <alica/test/TestContext.h>
+#include <alica/test/Util.h>
+#include <alica_tests/BehaviourCreator.h>
+#include <alica_tests/ConditionCreator.h>
+#include <alica_tests/ConstraintCreator.h>
+#include <alica_tests/PlanCreator.h>
+#include <alica_tests/TestWorldModel.h>
+#include <alica_tests/TransitionConditionCreator.h>
+#include <alica_tests/UtilityFunctionCreator.h>
+#include <clock/AlicaRosTimer.h>
+#include <communication/AlicaDummyCommunication.h>
+#include <logger/AlicaRosLogger.h>
+
+#include <gtest/gtest.h>
+#include <ros/ros.h>
+
+namespace alica::test
 {
 
-class AlicaSimplePlan : public AlicaTestFixture
+class TestSimplePlanFixture : public ::testing::Test
 {
+
+public:
+    virtual void SetUp() override
+    {
+        ros::NodeHandle nh;
+        std::string path;
+        nh.param<std::string>("/rootPath", path, ".");
+        _tc = std::make_unique<TestContext>("hairy", path + "/etc/", "Roleset", "SimpleTestPlan", false, 1);
+        ASSERT_TRUE(_tc->isValid());
+        const YAML::Node& config = _tc->getConfig();
+        _spinner = std::make_unique<ros::AsyncSpinner>(config["Alica"]["ThreadPoolSize"].as<int>(4));
+        _tc->setCommunicator<alicaDummyProxy::AlicaDummyCommunication>();
+        _tc->setTimerFactory<alicaRosTimer::AlicaRosTimerFactory>();
+        _tc->setLogger<alicaRosLogger::AlicaRosLogger>(config["Local"]["ID"].as<int>());
+        _tc->setWorldModel<alicaTests::TestWorldModelNew>(_tc.get());
+        _spinner->start();
+        AlicaCreators creators{std::make_unique<alica::ConditionCreator>(), std::make_unique<alica::UtilityFunctionCreator>(),
+                std::make_unique<alica::ConstraintCreator>(), std::make_unique<alica::BehaviourCreator>(), std::make_unique<alica::PlanCreator>(),
+                std::make_unique<alica::TransitionConditionCreator>()};
+        _tc->init(std::move(creators));
+    }
+
+    void TearDown() override
+    {
+        _spinner->stop();
+        _tc->terminate();
+    }
+
 protected:
-    const char* getRoleSetName() const override { return "Roleset"; }
-    const char* getMasterPlanName() const override { return "SimpleTestPlan"; }
-    bool stepEngine() const override { return false; }
+    std::unique_ptr<TestContext> _tc;
+    std::unique_ptr<ros::AsyncSpinner> _spinner;
 };
 
 /**
  * Tests whether it is possible to run a behaviour in a primitive plan.
  */
-TEST_F(AlicaSimplePlan, runBehaviourInSimplePlan)
+TEST_F(TestSimplePlanFixture, runBehaviourInSimplePlan)
 {
     ASSERT_NO_SIGNAL
-    ae->start();
+    _tc->startEngine();
+
     alica::AlicaTime sleepTime = alica::AlicaTime::seconds(1);
     uint8_t timeoutCount = 0;
-
     do {
-        ae->getAlicaClock().sleep(sleepTime);
-    } while (!alica::test::Util::isPlanActive(ae, 1412252439925));
+        _tc->sleep(sleepTime);
+    } while (!_tc->isPlanActive(1412252439925));
```

Comment:
Among the reasons for refactoring tests (which this PR violates) is to
- not use unreadable ids like 1412252439925 etc. for referring to plans, states, conditions etc. Instead we should use names. Therefore, you need to set the names in the designer first. 
- not use sleeps & instead use STEP_UNTIL
- not use statics like CounterClass::called etc. Instead need to use world model or blackboard
- try to simplify (or generalize) the logic of checking if a behaviour etc. is called instead of using weird counters & timeouts 
- Not to define too many test fixtures (can reuse the one that is already there in test_success)
This PR still has the above problems